projects/wvr-elements/src/lib/core/wvr-animation.service.ts

```diff
@@ -0,0 +1,107 @@
+import { animation, AnimationBuilder, AnimationMetadata, AnimationPlayer, AnimationReferenceMetadata, useAnimation } from '@angular/animations';
+import { Injectable } from '@angular/core';
+import { wvrAnimationDefaults, wvrAnimationInitialization, wvrAnimations } from '../shared/animation/wvr-animations';
+import { WvrBaseComponent } from '../shared/wvr-base.component';
+
+@Injectable({
+  providedIn: 'root',
+  deps: [AnimationBuilder]
+})
+export class WvrAnimationService {
+
+  private recieversRegistry = new Map<string, Array<WvrBaseComponent>>();
+
+  private animationStates = new Map<number, Map<string, boolean>>();
+
+  constructor(private readonly builder: AnimationBuilder) { }
+
+  registerAnimationReciever(recieverName: string, component: WvrBaseComponent): void {
+    let recievers = this.recieversRegistry.get(recieverName);
```

Comment:
Is receivers a good name for this? I have no idea what that means in this context and would expect to know from an explicit variable name.